packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualPositionManager.sol

```diff
@@ -160,13 +173,31 @@ contract PerpetualPositionManager is FeePayer, FundingRateApplier, Administratee
         require(_getIdentifierWhitelist().isIdentifierSupported(_priceIdentifier), "Unsupported price identifier");
 
         withdrawalLiveness = _withdrawalLiveness;
-        TokenFactory tf = TokenFactory(_tokenFactoryAddress);
-        tokenCurrency = tf.createToken(_syntheticName, _syntheticSymbol, 18);
+        tokenCurrency = ExpandedIERC20ExclusiveMinter(_tokenAddress);
         minSponsorTokens = _minSponsorTokens;
         priceIdentifier = _priceIdentifier;
         excessTokenBeneficiary = _excessTokenBeneficiary;
     }
 
+    /**
+     * @notice Must be called before any sponsor can mint new synthetic debt. This function
+     * should execute all prerequisite setup logic and run checks on the token currency.
+     * @dev Contract must have the Owner role for the `tokenCurrency`.
+     */
+    function initialize() public nonReentrant() {
```

Comment:
Is this only necessary because the token minter role must be assigned after the contract is deployed because we don't know the address? Is there a simple way to predict the address so we don't need this separate method?

The only reason I ask is that it seems like the checking of the token contract adds a significant amount of complexity (I wasn't aware it would be this difficult to add that check) in terms of bytecode and readability. Furthermore, nothing can really happen until create is first called, which would immediately blow up if the minter role wasn't already granted. If we can't move this to the constructor, thoughts on just removing the check and checking the total supply == 0 in the constructor?

Reply:
This is not necessary but important (imo) because the perpetual has no idea who has the owner or minter roles for the synthetic. So it's a safety guard to enforce that the perpetual is the only owner and only minter for the contract.

Reply:
We can definitely move the total supply check to the constructor or remove it completely. This check wasn't necessary with the EMP because the EMP creates the synthetic contract, and therefore "knows" its supply starts at 0

Reply:
I think the initialize function provides added security in an ideal setting, but we can also make it the first to be removed if we require additional bytecode space in a future PR